### PR TITLE
Bxmsdoc 7507 main

### PR DESCRIPTION
**Pull request by mramendi**

### CODE REVIEW

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc [hunk 1]
```diff
@@ -9,12 +9,12 @@ ifdef::PAM,DM[]
...
+* Create a start event, intermediate catch event, or boundary event (attached to a human task) of the `message` type. {KIE_SERVER} automatically subscribes to the Kafka topic that is defined in the message. A message triggers the event. The event node acts as the consumer of the message and can pass the content of the message to the subsequent node in the process.
 
 * Create an end event or intermediate throw event of the type `message`. When the process triggers the event, {KIE_SERVER} sends a Kafka message in the topic that is defined in the message. The message contains the data that is configured in the event. The event node acts as the producer of the message.
 
 * Add the `KafkaPublishMessages` custom task to the process. This task does not require the {KIE_SERVER} Kafka capability but can be more complicated to configure than message events.
 
-* Configure your service and {KIE_SERVER} to emit Kafka messages about every completed process, case, and task when transactions are committed.
+* Configure your service and {KIE_SERVER} to automatically post (emit) Kafka messages about every completed process, case, and task when transactions are committed.
```
Comment (csherrar) [inline doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc:20]: [question] - Is there a style rule for adding alternative words such as "emit" in parentheses?

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/kieserver-kafka-customformat-proc.adoc [hunk 2]
```diff
@@ -20,5 +20,17 @@ You can download the interface definitons from the https://github.com/kiegroup/d
...
-** `org.kie.server.jbpm-kafka.ext.eventReaderClass`: the custom event reader class. Set this property to use a different format to receive messages. If you want to use a custom format, set the property to the fully qualified name of your custom event reader class. If you want to use a raw JSON data format, set the property to `org.kie.server.services.jbpm.kafka.RawJsonEventReader`.
++
+.{KIE_SERVER} system properties for setting a custom writer or reader
+[cols="45%,55%", options="header"]
+|===
+|Property
+|Description
+
+|`org.kie.server.jbpm-kafka.ext.eventWriterClass`
+|The custom event writer class. Set this property to use a different format to send messages. If you want to use a custom format, set the property to the fully qualified name of your custom event writer class. If you want to use a raw JSON data format, set the property to `org.kie.server.services.jbpm.kafka.RawJsonEventWriter`.
```
Comment (csherrar) [inline doc-content/jbpm-docs/src/main/asciidoc/Kafka/kieserver-kafka-customformat-proc.adoc:31]: [Question] - Is "send", in this case, correct? Just considering the alternatives "post" and "emit".

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc (hunk 1 above)
Comment (csherrar) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc:20]: [Style] Having reread the section, maybe replacing "post" and "emit" with "send":

s/to automatically post (emit) Kafka messages/to automatically send Kafka messages

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-customtask-proc.adoc [hunk 3]
```diff
@@ -3,9 +3,13 @@
...
 You can add a `KafkaPublishMessages` custom task to your process. This task sends Kafka messages. It does not use the {KIE_SERVER} Kafka capability, so you can use this task in processes that do not run on a {KIE_SERVER}. However, this task can be more complicated to configure than other {KAFKA_PRODUCT} integration options.
 
+.Prerequisites
+
+* You are logged in to {CENTRAL} as an administrative user.
+
 .Procedure
 
-. In the {CENTRAL} administrative settings menu, as the administrative user, select *Custom Tasks Administration*.
+. In the {CENTRAL} administrative settings menu, select *Custom Tasks Administration*.
```
Comment (csherrar) [inline doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-customtask-proc.adoc:12]: [Structure] - This is a good change. I will look out for similar.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-receive-event-proc.adoc [hunk 4]
```diff
@@ -3,26 +3,31 @@
...
+. Add a start event, an intermediate catch event, or a boundary event that is attached to a human task. The event must have the `message` type.
 . Open the properties of the event.
-. In the *Message* field, select *New* and then enter the name of the message. This name must be the same as the name of the topic from which the event is to receive Kafka messages, or else must be defined in an `org.kie.server.jbpm-kafka.ext.topics._broker-topic-name_` {KIE_SERVER} system property.
+. In the *Message* field, select *New* and then enter the name of the message. This name must be the same as the name of the topic from which the event is to receive Kafka messages, or else must be defined as `_topic-name_` in an `org.kie.server.jbpm-kafka.ext.topics._topic-name_` {KIE_SERVER} system property.
 +
 For instructions about using `org.kie.server.jbpm-kafka.ext.topics.*` system properties to define topic names, see xref:kieserver-kafka-proc_{context}[].
 +
-. Add an output data item. Select the data object that you created as its type.
+. Add an output data item and set the type of this data item to the data object that you created.
```
Comment (csherrar) [inline doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-receive-event-proc.adoc:30]: [Style suggestion] - Rephrase to remove "of this" and "that you created":

s/Add an output data item and set the type of this data item to the data object that you created./Add an output data item and set the data type to match the data object.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-send-event-proc.adoc [hunk 5]
```diff
@@ -19,13 +24,13 @@ endif::JBPM,DROOLS,OP[]
 . Select the business process and open the business process designer.
 . Add an intermediate throw event or an end event of the type `message`.
 . Open the properties of the event.
-. In the *Message* field, select *New* and then enter the name of the message. This name must be the same as the name of the topic to which the event is to send Kafka messages, or else must be defined in an `org.kie.server.jbpm-kafka.ext.topics._broker-topic-name_` {KIE_SERVER} system property.
+. In the *Message* field, select *New* and then enter the name of the message. This name must be the same as the name of the topic to which the event is to send Kafka messages, or else must be defined as `_topic-name_` in an `org.kie.server.jbpm-kafka.ext.topics._topic-name_` {KIE_SERVER} system property.
 +
 For instructions about using `org.kie.server.jbpm-kafka.ext.topics.*` system properties to define topic names, see xref:kieserver-kafka-proc_{context}[].
 +
-. Add an input data item. Select the data object that you created as its type.
+. Add an input data item and set the type of this data item to the data object that you created.
```
Comment (csherrar) [inline doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-send-event-proc.adoc:31]: [Style suggestion] - Rephrase to remove "of this" and "that you created":

s/Add an input data item and set the type of this data item to the data object that you created./Add an input data item and set the data type to match the data object.

**Review by csherrar — COMMENTED** [review]
Some suggestions and questions, otherwise looks good to me!

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc [hunk 6]
```diff
@@ -9,12 +9,12 @@ ifdef::PAM,DM[]
 endif::PAM,DM[]
 is a streaming platform. It acts as a message broker, passing messages, which are sorted into topics, between applications in a software environment.
 
-Using {PRODUCT}, you can create business processes that send and receive Kafka messages in the following ways:
+Using {PRODUCT}, you can create business processes that send and receive Kafka messages. You can use the following methods to create such business processes:
```
[the remark this answers is not in the document]
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc:12]: I don't think this change works. This is not about methods of creating business processes in general, but about methods of creating business processes that send and receive Kafka messages.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc (hunk 1 above)
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc:20]: I have now fixed this issue by getting rid of "emit" but adding "configure with an emitter" instead.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-receive-event-proc.adoc (hunk 4 above)
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-receive-event-proc.adoc:30]: I don't think this change works. This is not just any data object, but the data object that the user created at the very start if the process, several steps earlier. We can not link the previous step, so at least we have to say "the data object that you created" explicitly.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-send-event-proc.adoc (hunk 5 above)
Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/message-send-event-proc.adoc:31]: Same comment as before.

File: doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (csherrar) [reply doc-content/jbpm-docs/src/main/asciidoc/Kafka/integration-kafka-con.adoc:12]: I understand what you mean.

Perhaps, "You can use the following methods to create business processes that send and receive Kafka messages."

**Pull request merged**